### PR TITLE
feat: remove optimize's implementation of backups

### LINKED ISSUE #24476: Remove current implementation of Backups from Tasklist, Operate and Optimize
As we are centralizing the backups of Tasklist, Identity, Operate and Optimize we should remove the current implementation of backups for those components.

### Acceptance Criteria
- [x] As part of 8.7 there is no Tasklist specific implementation of backups
- [x] As part of 8.7 there is no Operate specific implementation of backups
- [ ] As part of 8.7 there is no Optimize specific implemenation of backups (here we have the risk of Optimize not making to be part of single-application as part of 8.7, please double check in [#prj-optimize-as-part-of-single-application](https://camunda.slack.com/archives/C07MMF3MRPA)

### PR DESCRIPTION
**Pull request by entangled90**
## Description
Removes all code related to backups in optimize, since the implementation is now common to all webapps
## Checklist

- [x] Wait for #26196 to be merged

## Related issues

closes #24476

### CODE REVIEW

File: optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java [hunk 1]
```diff
@@ -77,7 +76,6 @@ public class OptimizeTomcatConfig {
             OptimizeResourceConstants.STATIC_RESOURCE_PATH,
             OptimizeResourceConstants.ACTUATOR_ENDPOINT,
             PanelNotificationConstants.SEND_NOTIFICATION_TO_ALL_ORG_USERS_ENDPOINT,
-            RestConstants.BACKUP_ENDPOINT,
```
Comment (entangled90) [inline optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java:80]: Should endpoints defined in actuator in the `dist` module be listed here or not?

**Review by grlimacan — CHANGES_REQUESTED** [review]
Hi @entangled90 , the removals you did are ok, but I believe there are still quite a few places where things can be removed. Here's a list of what I have found so far:

- AbstractDatabaseSchemaTestClient: everything related to snapshots
- ElasticSearchConfiguration / OpenSearchConfiguration/service-config.yaml / ConfigurationServiceConstants / ConfigurationServiceTest: Quite a few configs about Backup present - note it might be a wise choice to keep those though, because removing there might be a breaking change for existing customers (needs to be tested). If that's the case, please document accordingly
- DatabaseBackup.java can be removed

**Comment by entangled90** [comment]
- `AbstractDatabaseSchemaTestClient` it's used by `Upgrade{Opensearch/Elasticsearch}SchemaIT` for example, so I would not remove it
- `DatabaseBackup` is still used atm from the common `dist` module (as optimize is still able to take backup, but it's just using shared code)

File: optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java (hunk 1 above)
Comment (entangled90) [reply optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java:80]: @grlimacan is removing this ok? should I add another rule for "backupHistory" (a new endpoint that was added) ?

Comment (grlimacan) [reply optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java:80]: I am actually not famililar with the new TomcatConfig, @buccarel is probably the one that can answer this question best

**Comment by grlimacan** [comment]
> * AbstractDatabaseSchemaTestClient

@entangled90 , here I meant not to remove the whole file, but all the parts that are related to snapshots. I'm also fine with keeping this for the moment, as long as there is a follow up ticket for it

**Review by grlimacan — APPROVED** [review]
Pre-approving, necessary changes (if any) are not critical

**Comment by entangled90** [comment]
In the `Elasticsearch/OpenSearchSchemaTestClient`  a backup/restore was performed before upgrading the schema. I removed that part as it didn't feel directly related to the upgrade and then removed all the snapshot related methods that were not in used.

**Review by grlimacan — APPROVED** [review]
Happy for you to merge when pipeline is green

File: optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java (hunk 1 above)
Comment (buccarel) [reply optimize/backend/src/main/java/io/camunda/optimize/OptimizeTomcatConfig.java:80]: From what I'm understanding, the handler of such new endpoint is already implemented outside of the optimize module. This should mean that there's no need to declare the endpoint in optimize

**Pull request merged**